step-certificates/templates/policy.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if .Values.networkpolicy.enabled -}}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ include "step-certificates.fullname" . }}-policy
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "step-certificates.labels" . | nindent 4 }}
+  {{- with .Values.networkpolicy.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  podSelector: 
+    matchLabels:
+      app.kubernetes.io/name: {{ include "step-certificates.name" . }}
+  policyTypes:
+  - Ingress
+  - Egress
+  ingress:
+  - ports:
+    - protocol: TCP
+      port: {{ .Values.service.targetPort }}
+    - protocol: TCP
+      port: {{ .Values.service.port }}
```

Comment:
I'm not familiar with networkPolicy, but isn't the targetPolicy enough? 

`.Values.service.port` is used by the ingress (ingress.yaml). I'm unsure if it should be here, as the selector matches the pod. If it is necessary, it should be only there if the ingress is enabled.

Reply:
Ah I see, you think we won't need `.Values.service.Port` (which is on the ingres and service) because we go in via the `.Values.service.targetPort` (on the container), or maybe vice-versa?  It might be that we can get rid of targetPort if either Service or Ingres are enabled, and get rid of Port if neither are. TBH I'm not 100% certain how kubes works on this, so I opened both to be safe (as its no extra risk if the request bypasses the ingres).  If anyone has a definite on this Im happy for guidance?